Mathlib/Tactic/PPWithUniv.lean

```diff
@@ -0,0 +1,40 @@
+/-
+Copyright (c) 2023 Microsoft Corporation. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Gabriel Ebner
+-/
+import Lean
+
+/-!
+# Command to pretty-print universe level parameters by default
+
+This module contains the `pp_with_univ` command, which enables pretty-printing
+of universe parameters for the specified definition.  This is helpful for definitions like
+`Ordinal`, where the universe levels are both relevant and not deducible from the arguments.
+-/
+
+namespace Mathlib.PPWithUniv
+
+open Lean PrettyPrinter Delaborator SubExpr Elab Command
+
+/--
+Delaborator that prints the current application with universe parameters on the head symbol,
+unless `pp.universes` is explicitly set to `false`.
+-/
+def delabWithUniv : Delab :=
+  whenPPOption (·.get pp.universes.name true) <|
```

Comment:
I didn't realize boolean options were tri-state, but I guess that makes sense given how `pp.all` is implemented as providing default values for other pretty printer options.

bors merge